lighthouse-extension/test/extension-test.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env mocha */
+
+const path = require('path');
+const assert = require('assert');
+const fs = require('fs');
+const puppeteer = require('puppeteer');
+
+const lighthouseExtensionPath = path.resolve(__dirname, '../app');
+const config = require(path.resolve(__dirname, '../../lighthouse-core/config/default.js'));
+const lighthouseCategories = Object.keys(config.categories);
+
+let extensionPage;
+
+const assertAuditElements = async ({category, expected, selector}) => {
+  const elementCount = await extensionPage.evaluate(({category, selector}) =>
+    document.querySelector(`#${category}`).parentNode.querySelectorAll(selector).length,
+    {category, selector}
+  );
+
+  assert.equal(expected, elementCount,
+    `${category} does not have the correct amount of audits`);
+};
+
+const assertReport = async () => {
+  const categories = await extensionPage.$$(`#${lighthouseCategories.join(',#')}`);
+  assert.equal(categories.length, lighthouseCategories.length,
+    `${categories.join(' ')} does not match ${lighthouseCategories.join(' ')}`);
+
+  for (const category of lighthouseCategories) {
+    let selector = '.lh-audit';
+    let expected = config.categories[category].audits.length;
+    if (category === 'performance') {
+      selector = '.lh-audit,.lh-timeline-metric,.lh-perf-hint,.lh-filmstrip';
+      // we deducted the expected by 1 because network-requests audit is not showing in the report
+      expected = expected - 1;
```

Comment:
i think what we want here is more like

```js
expected = config.categories[category].audits.map(a => !!a.group).length;
```

Reply:
filmstrip doesn't have a group either so I'll lose that one too. I could test for that separately.`

```
{id: 'screenshot-thumbnails', weight: 0},
{id: 'network-requests', weight: 0},
```

Reply:
good catch. IMO let's remove filmstrip from the selector above and use this `.length`.